crossbeam-skiplist/src/map.rs

```diff
@@ -248,6 +249,20 @@ impl<'a, K, V> Entry<'a, K, V> {
     }
 }
 
+impl<'a, K, V> Drop for Entry<'a, K, V>
+{
+    fn drop(&mut self) {
+        let guard = &epoch::pin();
+        let dummy: [usize; 2] = [1; 2]; // 1 because a refrence to 0 will crash :)
+        unsafe {
+            // Safe because the size of `RefEntry` is same as two pointers,
+            // which in turn is equal to two `usize`s.
+            let transmuted = mem::transmute::<[usize; 2], base::RefEntry<K, V>>(dummy);
+            mem::replace(&mut self.inner, transmuted).release(guard);
```

Comment:
Perhaps a cleaner way to do this would be to wrap `inner` into a `ManuallyDrop` and then here do something like:

```rust
let guard = &epoch::pin();
unsafe {
    ManuallyDrop::take(&mut self.inner).release(guard);
}
```

Reply:
Thanks, I was not aware of `ManuallyDrop`, seems like a great fit here.

`ManuallyDrop::take` is nightly-only API, so I have added a feature attribute to enable it.